AEXOL Pycord Switch Project

KnightChaser opened the pull request:
Basic setup and changed commands ping (2 types/specified target and bot(latency check)) as a slash command


**Previous features to be switched above all** and current status
- [ ] Alive check (Hello)
- [ ] Exception check
- [ ] Bot status change (if it's still necessary)
- [ ] Command authorization and permission check
- [ ] Help feature
- [ ] Get time
- [ ] Ping to the specified target (+Bot latency check)
- [ ] Real-time cryptocurrencies information fetch
- [ ] KRX major indexes fetching
- [ ] Google index searching

KnightChaser committed:
First pycord switch trial

Basic setup and changed commands ping and get_time as slash commands

KnightChaser committed:
2 of 9 switched to pycord

Changed command get_time as a slash command

**Previous features to be switched above all** and current status
- [ ] Alive check (Hello)
- [ ] Exception check
- [ ] Bot status change (if it's still necessary)
- [ ] Command authorization and permission check
- [ ] Help feature
- [x] Get time
- [x] Ping to the specified target (+Bot latency check)
- [ ] Real-time cryptocurrencies information fetch
- [ ] KRX major indexes fetching

KnightChaser committed:
3 of 9 switched to pycord

Changed command get_time as a slash command

**Previous features to be switched above all** and current status
- [x] Alive check (Hello)
- [ ] Exception check
- [ ] Bot status change (if it's still necessary)
- [ ] Command authorization and permission check
- [ ] Help feature
- [x] Get time
- [x] Ping to the specified target (+Bot latency check)
- [ ] Real-time cryptocurrencies information fetch
- [ ] KRX major indexes fetching

KnightChaser committed:
4 of 10 switched to pycord (+Google index search)

Changed command google as a slash command

**Previous features to be switched above all** and current status
- [x] Alive check (Hello)
- [ ] Exception check
- [ ] Bot status change (if it's still necessary)
- [ ] Command authorization and permission check
- [ ] Help feature
- [x] Get time
- [x] Ping to the specified target (+Bot latency check)
- [ ] Real-time cryptocurrencies information fetch
- [ ] KRX major indexes fetching
- [x] Google index searching

x3onkait commented:
At the time when I was creating the branch to switch this bot to `pycord`-based, Accidentally I missed the `Google indexing search` features to be switched in the to-do list, That's why the title of the branch push has changed from ` ... of 9` to ` ... of 10`. 

The first comment of this branch has been edited by applying this fix.

KnightChaser committed:
Add extra feature + calender.month

Just add monthly calender feature as an extra feature, with interacting Discord slash commands feature

KnightChaser committed:
5 of 10 switched to pycord (+Crypto info search)

Changed command google as a slash command

**Previous features to be switched above all** and current status
- [x] Alive check (Hello)
- [ ] Exception check
- [ ] Bot status change (if it's still necessary)
- [ ] Command authorization and permission check
- [ ] Help feature
- [x] Get time
- [x] Ping to the specified target (+Bot latency check)
- [x] Real-time cryptocurrencies information fetch
- [ ] KRX major indexes fetching
- [x] Google index searching

x3onkait commented:
I made some changes to the cryptocurrencies feature. (`cogs/finance.py`)

- All contents will be served with identical
- I thought the crypto is related to the **finance** sector so I annexed this feature to the `finance` slash command. So now we retrieve the information with `/finance crypto_market` and `/finance crypto_rank`

```python
finance = discord.SlashCommandGroup(name = "finance", description = "금융 정보 조회")


    @finance.command(description = "암호화폐 시세 조회")
    async def crypto_market(self,
                            ctx,
                            fiat_currency: Option(str, "거래통화(마켓)", choices = ["KRW", "BTC", "USDT"]),
                            crypto_symbol: Option(str, "조회할 암호화폐 종목(ex. BTC)")):

        market_code = f"{fiat_currency}-{crypto_symbol}"
        result_as_embed = await get_crypto_info(ctx, market_code)
        await ctx.respond(embed = result_as_embed)

    @finance.command(description = "암호화폐 통계 조회")
    async def crypto_rank(self,
                          ctx,
                          fiat_currency: Option(str, "거래통화(마켓)", choices = ["KRW", "BTC", "USDT"]),
                          sort_criteria: Option(str, "조회 기준", choices = ["시가(Opening Price)", "고가(High Price)", "저가(Low Price)", "현재가(Trade Price)",
                                                                            "전일 종가(UTC 0시 기준)", "시세 변화액(24시간)", "시세 변화율(24시간)",
                                                                            "누적 거래대금(UTC 0시 기준)", "누적 거래대금(24시간)"]),
                          sort_direction: Option(str, "방향", choices = ["오름차순(ascending)", "내림차순(descending)"])):

        ...

        result = await get_crypto_ranking_by(ctx, market = fiat_currency,
                                                  criteria = sort_criteria_conversion[sort_criteria],
                                                  order = sort_direction_conversion[sort_direction])
        await ctx.respond(result)
```
